src/DBFTables.jl

```diff
@@ -1,150 +1,293 @@
 module DBFTables
 
-using DataFrames, Printf
-
-# Read DBF files in xBase format
-# Files written in this format have the extension .dbf
-# Implemented: dBase III+ (w/o memo)
-
-struct DBFFieldDescriptor
-	nam::String
-	typ::Type
-	len::Int8
-	dec::Int8
-end
-
-struct DBFHeader
-	version::UInt8
-	lastUpdate::String
-	records::Int32
-	hsize::Int16
-	rsize::Int16
-	incomplete::Bool
-	encrypted::Bool
-	mdx::Bool
-	langId::UInt8
-	fields::Vector{DBFFieldDescriptor}
-end
-
-function dbf_field_type(fld::Char, dec::UInt8)
-	rt = Nothing
-	if fld == 'C'
-		rt = String
-	elseif fld == 'D'
-		rt = String
-	elseif fld == 'N'
-		if dec > 0
-			rt = Float64
-		else
-			rt = Int
-		end
-	elseif fld == 'F' || fld == 'O'
-		rt = Float64
-	elseif fld == 'I' || fld == '+'
-		rt = Integer
-	elseif fld == 'L'
-		rt = Bool
-	else
-		 warn("Unknown record type: $(fld)")
-	end
-	return rt
+using Printf, Tables, WeakRefStrings
+
+"Field/column descriptor, part of the Header"
+struct FieldDescriptor
+    name::Symbol
+    type::DataType
+    length::UInt8
+    ndec::UInt8
+end
+
+"DBF header, which also holds all field definitions"
+struct Header
+    version::UInt8
+    last_update::String
+    records::UInt32
+    hsize::UInt16
+    rsize::UInt16
+    incomplete::Bool
+    encrypted::Bool
+    mdx::Bool
+    lang_id::UInt8
+    fields::Vector{FieldDescriptor}
+end
+
+"Struct representing the DBF Table"
+struct Table
+    header::Header
+    data::Vector{UInt8}  # WeakRefString references this
+    strings::StringArray{WeakRefString{UInt8},2}
```

Comment:
Any reason not to make this `StringArray{String, 2}`? I'd feel better with that extra bit of safety (i.e. with `StringArray{WeakRefString{UInt8}, 2}`, if the user somehow indexes a single `WeakRefString`, then the parent StringArray/Table get garbage-collected, then their `WeakRefString` will be invalid. So it's safer to use `String` so the String gets allocated when indexed)